File: src/server/Shared/Shared.DTOs/Enums/PayslipType.cs
namespace FluentPOS.Shared.DTOs.Enums
{
    /// <summary>
    /// ATTENDENCE BASED:EMPLOYEES HAVE TO MARK THEIR ATTENDANCE IN A DAY DESPITE COMPLETING THEIR REQUIRED HOURS.
    /// TIME BASED: EMPLOYEE HAVE TO COMPLETE THEIR ASSIGNED HOURS WITHIN SHIFT-TIME. EG: 9:00 Am TO 5:00 Pm
    /// HOUR BASED:EMPLOYEES HAVE TO COMPLETE THEIR REQUIRED HOURS ANYTIME WITHIN 24HRS.
    /// </summary>
    public enum PayslipType
    {
        /// <summary>
        /// EMPLOYEES HAVE TO MARK THEIR ATTENDANCE IN A DAY DESPITE COMPLETING THEIR REQUIRED HOURS.
        /// </summary>
        AttendanceBase = 1,

        /// <summary>
        /// EMPLOYEE HAVE TO COMPLETE THEIR ASSIGNED HOURS WITHIN SHIFT-TIME.
        /// </summary>
        TimeBase = 2,

        /// <summary>
        /// EMPLOYEES HAVE TO COMPLETE THEIR REQUIRED HOURS ANYTIME WITHIN 24HRS.
        /// </summary>
        HourlyBase = 3
    }
}

File: src/client/src/app/core/enums/PayslipType.ts
export enum PayslipType {
    AttendanceBase = 1,
    TimeBase = 2,
    HourlyBase = 3
}
export const PayslipTypeMapping: Record<PayslipType, string> = {
    [PayslipType.AttendanceBase]: "Attendance Base",
    [PayslipType.TimeBase]: "Time Base",
    [PayslipType.HourlyBase]: "Hourly Base"
};
